src/lsm/manifest.zig

```diff
@@ -219,14 +229,41 @@ pub fn ManifestType(comptime Table: type, comptime Storage: type) type {
             const manifest_level = &manifest.levels[level];
 
             assert(table.snapshot_max >= snapshot);
-            manifest_level.set_snapshot_max(snapshot, table);
+            manifest_level.set_snapshot_max(snapshot, .{}, table);
             assert(table.snapshot_max == snapshot);
 
             // Append update changes to the manifest log
             const log_level = @intCast(u7, level);
             manifest.manifest_log.insert(log_level, table);
         }
 
+        pub fn move_table(
+            manifest: *Manifest,
+            level_a: u8,
+            level_b: u8,
+            snapshot: u64,
+            table: *TableInfo,
+        ) void {
+            const manifest_level_a = &manifest.levels[level_a];
+            const manifest_level_b = &manifest.levels[level_b];
+
+            assert(table.snapshot_max >= snapshot);
+            if (constants.verify) {
+                assert(manifest_level_a.contains(table));
+            }
+
+            // Insert the table into level b.
+            manifest_level_b.insert_table(manifest.node_pool, table);
+            manifest.manifest_log.insert(@intCast(u7, level_b), table);
+
+            // Update the table in level a, marking it explicitely as "moved" to prevent its blocks
+            // from being collected as they're now referenced by the inserted table in level b.
+            manifest_level_a.set_snapshot_max(snapshot, .{ .moved = true }, table);
```

Comment:
I don't think we should be tracking `.moved` permanently as a flag in the Manifest, simply in order to decide whether to collect blocks here. The former is long-lived, but this decision is short-lived. We also then don't need to introduce the Flags type yet.

Reply:
What is an alternative way to ensure that "moved" tables with updated snapshots don't get marked as removed to the `ManifestLog` during `remove_invisible_tables()`? The intrusive approach seemed simplest, but we could instead keep a separate hashmap/list that's checked in `remove_invisible_tables()`. Issue there is that the capacity would have to be adjusted with any manifest access changes to Compaction or size changes to higher components like Tree and Groove.

Reply:
Ended up implementing it using a separate buffer. It takes advantage of there only being one moved table for each level until a matching call to `remove_invisible_tables()`.